packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx

```diff
@@ -1,63 +1,86 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 
 import { send, sendCatch } from 'loot-core/platform/client/fetch/index';
+import { addNotification } from 'loot-core/src/client/actions';
+import { calculateHasWarning } from 'loot-core/src/client/reports';
 import * as monthUtils from 'loot-core/src/shared/months';
 import { type CustomReportEntity } from 'loot-core/types/models/reports';
 
 import { useAccounts } from '../../../hooks/useAccounts';
 import { useCategories } from '../../../hooks/useCategories';
 import { useLocalPref } from '../../../hooks/useLocalPref';
 import { usePayees } from '../../../hooks/usePayees';
-import { useResponsive } from '../../../ResponsiveProvider';
+import { SvgExclamationSolid } from '../../../icons/v1';
 import { styles } from '../../../style/index';
 import { theme } from '../../../style/theme';
 import { Block } from '../../common/Block';
+import { InitialFocus } from '../../common/InitialFocus';
+import { Input } from '../../common/Input';
 import { Text } from '../../common/Text';
+import { Tooltip } from '../../common/Tooltip';
 import { View } from '../../common/View';
+import { NON_DRAGGABLE_AREA_CLASS_NAME } from '../constants';
 import { DateRange } from '../DateRange';
 import { ReportCard } from '../ReportCard';
 
 import { GetCardData } from './GetCardData';
-import { ListCardsPopover } from './ListCardsPopover';
+import { MissingReportCard } from './MissingReportCard';
 
-function index(data: CustomReportEntity[]): { [key: string]: boolean }[] {
-  return data.reduce((carry, report) => {
-    const reportId: string = report.id === undefined ? '' : report.id;
+type CustomReportListCardsProps = {
+  isEditing?: boolean;
+  report?: CustomReportEntity;
+  onRemove: () => void;
+};
 
-    return {
-      ...carry,
-      [reportId]: false,
-    };
-  }, []);
+export function CustomReportListCards({
+  isEditing,
+  report,
+  onRemove,
+}: CustomReportListCardsProps) {
+  // It's possible for a dashboard to reference a non-existing
+  // custom report
+  if (!report) {
```

Comment:
Instead of showing a placeholder 'missing' card, why not just remove it?

Reply:
It has the potential to mess up your layout if we automatically remove things from it. Which is not ideal.